lessons/advanced/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: advanced
+order: 9
+lang: en
+---
+
+If you want to define 'interface' like in Java or Ruby, Elixir contains syntax for that. More, if you want to define your custom type you can do that too. In this lesson we will learn about `@spec` and `@type` syntax.
```

Comment:
Let's get ride of this `think of ruby or java` mentality. I think this is a problem in the Elixir community where we keep making assumptions about each other and other languages.

It would be nice to replace this with something like your sentence on line 44: `Unlike documentation that is readable and understandable only for other humans (if only exists and is good written), `@spec` use more formal syntax and could be understand by machine.`